controllers/serving/inference_controller.go

```diff
@@ -273,6 +294,80 @@ func (ir *InferenceReconciler) syncTrafficDistribution(inf *servingv1alpha1.Infe
 	return nil
 }
 
+// syncPredictorTrafficDistribution syncs traffic-routing resources to ensure traffic is distributed by the
+// user configuration required.
+// For example, user A serves two models and distribute 90% traffic to model A.1 and the left to
+// model A.2, the network topology looks like this:
+//
+// User
+//  |---request---> VirtualService
+//                        |--- 90% ---> Deploy-Of-Model-A.1
+//                        |--- 10% ---> Deploy-Of-Model-B.1
+func (ir *InferenceReconciler) syncPredictorTrafficDistribution(inf *servingv1alpha1.Inference, index int, predictor *servingv1alpha1.PredictorSpec) error {
```

Comment:
can we make it pluggable such that both istio and caddy based traffic control can be configured. @SimonCqk

Reply:
yes, I think that would be better, it may require a global configuration